File: 백준/Bronze/5622. 다이얼/다이얼.py
call = {'': 1, 'ABC': 2, 'DEF': 3, 'GHI': 4, 'JKL': 5, 'MNO': 6, 'PQRS': 7, 'TUV': 8,
        'WXYZ': 9}

s = input()
time = 0
keys = list(call.keys())
for i in s:
    for j in keys:
        if i in j:
            time += call[j]

print(time + len(s))

File: 백준/Bronze/5622. 다이얼/다이얼.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;

public class Main {

	public static void main(String[] args) throws Exception {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String dial = br.readLine();

		Map<Character, Integer> map = new HashMap<>();
		for (char c : "ABC".toCharArray()) map.put(c, 2);
		for (char c : "DEF".toCharArray()) map.put(c, 3);
		for (char c : "GHI".toCharArray()) map.put(c, 4);
		for (char c : "JKL".toCharArray()) map.put(c, 5);
		for (char c : "MNO".toCharArray()) map.put(c, 6);
		for (char c : "PQRS".toCharArray()) map.put(c, 7);
		for (char c : "TUV".toCharArray()) map.put(c, 8);
		for (char c : "WXYZ".toCharArray()) map.put(c, 9);


		int ans = 0;
		for (char ch : dial.toCharArray()) {
			if (map.containsKey(ch)) {
				ans += map.get(ch) + 1;
			}
		}
		
		System.out.println(ans);

	}

}